### ISSUE TITLE
CA1508 false-positive when using compound assignments and ternary expressions

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn-analyzers#6532]
### Analyzer

**Diagnostic ID**: [CA1508](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.201](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

CA1508 produces false-positives when the modification is a compound assignment with a ternary expression.

### Steps To Reproduce

```csharp
void M()
{
    var i = 0;
    i += M2() ? 1 : 0;
    _ = i != 0; // CA1508: 'i != 0' is always 'false'. Remove or refactor the condition(s) to avoid dead code.
}

bool M2() => true;
```

### Expected behavior
No diagnostic.

### Actual behavior
CA1508 recognizes the expression as "always false", even though it may be true if `M2` returns `true`.

## Additional context
* If the compound assignment's RHS is a simple expression (e.g. invocation of a method returning an integer), the diagnostic is not emitted.
* If the compound assignment is converted into a regular assignment with a binary expression, the diagnostic is not emitted.

**Comment by CollinAlpert** [comment]
This took quite a bit of debugging, but I managed to make the test pass. It seems that the `DataFlowOperationVisitor` does not handle `IFlowCaptureReferenceOperation`s in compound assignments, only in simple assignments.
@buyaa-n Since this is deep in the ValueContentAnalysis, an area I am not at all familiar in, could you verify that my change does not have any obvious side-effects? Commit is here: https://github.com/dotnet/roslyn-analyzers/commit/8b41e0bd6d05e2e27df3fe6507b45c3c910ab1ba

**Comment by buyaa-n** [comment]
> Since this is deep in the ValueContentAnalysis, an area I am not at all familiar in, could you verify that my change does not have any obvious side-effects? Commit is here: https://github.com/dotnet/roslyn-analyzers/commit/8b41e0bd6d05e2e27df3fe6507b45c3c910ab1ba

Thanks @CollinAlpert, tagging @mavasani for flow analysis updates. Maybe better to just create a PR with the changes
